scylla/src/client/caching_session.rs

```diff
@@ -247,9 +247,129 @@ where
     }
 }
 
+const DEFAULT_MAX_CAPACITY: usize = 128;
+
+/// [CachingSessionBuilder] is used to create new [CachingSession] instances.
+///
+/// # Example
+///
+/// ```
+/// # use scylla::client::session::Session;
+/// # use scylla::client::session_builder::SessionBuilder;
+/// # use scylla::client::caching_session::{CachingSession, CachingSessionBuilder};
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+/// let session: Session = SessionBuilder::new()
+///     .known_node("127.0.0.1:9042")
+///     .build()
+///     .await?;
+/// let caching_session: CachingSession = CachingSessionBuilder::new(session)
+///     .max_capacity(2137)
+///     .build();
+/// # Ok(())
+/// # }
+/// ```
+pub struct CachingSessionBuilder<S = RandomState>
+where
+    S: Clone + BuildHasher,
+{
+    session: Session,
+    max_capacity: usize,
+    hasher: S,
+}
+
+impl CachingSessionBuilder<RandomState> {
+    /// Wraps a [Session] and creates a new [CachingSessionBuilder] instance,
+    /// which can be used to create a new [CachingSession].
+    ///
+    pub fn new(session: Session) -> Self {
+        Self {
+            session,
+            max_capacity: DEFAULT_MAX_CAPACITY,
+            hasher: RandomState::default(),
+        }
+    }
+}
+
+impl<S> CachingSessionBuilder<S>
+where
+    S: Clone + BuildHasher,
+{
+    /// Configures maximum capacity of the prepared statements cache.
+    pub fn max_capacity(mut self, max_capacity: usize) -> Self {
+        self.max_capacity = max_capacity;
+        self
+    }
+
```

Comment:
My one worry is that max capacity no longer has to be stated by the user explicitly, which may bite users unaware of this who use lots of different statements.

WDYT of holding Option<usize> and making build() return Result (Err if capacity was not specified)?

Reply:
Nah, this is ugly. At most, I could agree to accept `max_capacity` as a mandatory parameter in `CachingSessionBuilder::new()`.
But still - users should not expect that default configuration will work perfectly in their particular settings. Instead, we can warn them in a doc comment that they may want to tune `max_capacity` option.

Reply:
> Instead, we can warn them in a doc comment that they may want to tune max_capacity option.

Ok let's do that.

Reply:
Done.